eng/scripts/Download-DevOpsRecordings.ps1

```diff
@@ -0,0 +1,63 @@
+[CmdletBinding()]
+param ([Parameter(Mandatory=$True)][int] $PR, [switch] $NoWait = $false)
+
+$repoRoot = Resolve-Path "$PSScriptRoot/../..";
+$artifactsPath = Join-Path $repoRoot "artifacts"
+
+$token = (az account get-access-token --resource=https://management.core.windows.net/ -o json | ConvertFrom-Json).accessToken
+
+Write-Host "Processing builds for PR $PR"
+$builds = az pipelines runs list --organization https://dev.azure.com/azure-sdk --tags Recording --project internal --branch "refs/pull/$PR/merge" --query-order FinishTimeDesc -o json --only-show-errors | ConvertFrom-Json;
+
+$allCompleted = $true;
+do
+{
+    $builds = az pipelines runs list --organization https://dev.azure.com/azure-sdk --tags Recording --project internal --branch "refs/pull/$PR/merge" --query-order FinishTimeDesc -o json --only-show-errors | ConvertFrom-Json;
+
+    foreach ($build in $builds)
+    {
+        if ($build.status -ne "completed")
+        {
+            Write-Host "Waiting for '$($build.definition.name)' to finish - https://dev.azure.com/azure-sdk/internal/_build/results?buildId=$($build.id)"
+            $allCompleted = $false;
+        }
+    }
+
+    if (!$allCompleted)
+    {
+        Write-Host "Retrying after 10 seconds"
+        Start-Sleep -s 10
```

Comment:
Is 10s really the right number?  Or is the idea we'll wait until we think everything should more or less be done before fetching?

Reply:
Yeah, waiting for all runs to finish before fetching results.